src/libs/ajax/ajax-common.ts

```diff
@@ -62,22 +62,29 @@ export async function makeRequestRetry(request: Function, retryCount: number, ti
   }
 }
 
+const isUnauthorizedResponse = (error): boolean => error instanceof Response && error.status === 401;
+
+export const sessionTimedOutErrorMessage =
+  'Session timed out (due to auth token refresh failure or expired refresh token)';
+
 export const withRetryAfterReloadingExpiredAuthToken =
   (wrappedFetch: FetchFn): FetchFn =>
-  async (resource: RequestInfo | URL, options?: RequestInit) => {
+  async (resource: RequestInfo | URL, options?: RequestInit): Promise<Response> => {
     const requestHasAuthHeader = _.isMatch(authOpts(), options as object);
     try {
       return await wrappedFetch(resource, options);
     } catch (error) {
-      const isUnauthorizedResponse = error instanceof Response && error.status === 401;
-      if (isUnauthorizedResponse && requestHasAuthHeader) {
-        const successfullyReloadedAuthToken = !!(await reloadAuthToken());
+      if (isUnauthorizedResponse(error) && requestHasAuthHeader) {
+        const reloadedAuthTokenState = await loadAuthToken();
+        const successfullyReloadedAuthToken = !!reloadedAuthTokenState;
         if (successfullyReloadedAuthToken) {
           const optionsWithNewAuthToken = _.merge(options, authOpts());
           return await wrappedFetch(resource, optionsWithNewAuthToken);
         }
-        signOutAfterSessionTimeout();
-        throw new Error('Session timed out');
+        const causes =
```

Comment:
```suggestion
        const signOutCause =
```

Nit: the cause is singular, not plural. Also, it would be nice to specify what the cause is referring to.